### PR TITLE
add SyncMessages method to shhext api

### PR DESCRIPTION
**Pull request by adambabik**
This change adds a new JSON-RPC method `shhext_syncMessages`. It triggers syncing of historic messages between mail servers.

Closes https://github.com/status-im/status-go/issues/1290

## Changes

- added `shhext_syncMessages` with pagination support; this call is blocked until the response is received,
- e2e test that verifies `shhext_syncMessages` and pagination,
- 

## Todo

- [x] unit tests,
- [x] request params validations,
- [x] e2e tests,
- [x] providing mail sync response to the caller,
- [x] handling pagination,
- [x] update whisper after https://github.com/status-im/whisper/pull/15 is merged.

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [x] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?

**Comment by adambabik** [comment]
CI will keep failing till https://github.com/status-im/whisper/pull/15 is merged.

File: services/shhext/api.go [hunk 1]
```diff
@@ -105,6 +102,43 @@ func (r *MessagesRequest) setDefaults(now time.Time) {
...
+}
+
+// SyncMessagesResponse is a response from the mail server
+// to which SyncMessagesRequest was sent.
+type SyncMessagesResponse struct {
+	// Cursor from the response can be used to retrieve more messages
+	// for the previous request.
+	Cursor string `json:"cursor"`
+
+	// Error indicates that something wrong happen when sending messages
```
Comment (pedropombeiro) [inline services/shhext/api.go]:
```suggestion
	// Error indicates that something wrong happened when sending messages
```

File: t/helpers/retry.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+	for {
+		tries++
+
+		if err := fn(); err == nil {
+			return nil
+		}
+
+		if tries > maxRetries {
+			return errors.New("too manny retries")
```
Comment (pedropombeiro) [inline t/helpers/retry.go]:
```suggestion
			return errors.New("too many retries")
```

**Review by pedropombeiro — APPROVED** [review]
lgtm

**Pull request merged**